impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultSourceRoot.java

```diff
@@ -79,7 +79,12 @@ public DefaultSourceRoot(final Session session, final Path baseDir, final Source
         if (value != null) {
             directory = baseDir.resolve(value);
         } else {
-            directory = baseDir.resolve("src").resolve(scope.id()).resolve(language.id());
+            Path src = baseDir.resolve("src");
+            if (moduleName != null) {
+                directory = src.resolve(language.id()).resolve(moduleName).resolve(scope.id());
+            } else {
+                directory = src.resolve(scope.id()).resolve(language.id());
+            }
```

Comment:
```suggestion
            Path src = baseDir.resolve("src");
            directory = moduleName != null ?
            src.resolve(language.id()).resolve(moduleName).resolve(scope.id()) :
            src.resolve(scope.id()).resolve(language.id());
```

Reply:
Will wait to see what is the agreement on the default repository, because it would change that code.

Reply:
Changed the default to `src/${module}/${scope}/${lang}` which seems to be the preferred layout in the [thread on the developer mailing list](https://lists.apache.org/thread/dzvpyfr667z0bwmwn0tdjn6nlmnx1gcx) (choice 1 in the initial email).